tests/backup/backup_delete_test.go

```diff
@@ -0,0 +1,371 @@
+	"github.com/portworx/torpedo/drivers/backup"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/log"
+	. "github.com/portworx/torpedo/tests"
+)
+
+// IssueDeleteOfIncrementalBackupsAndRestore Issue delete of incremental backups in between and try to restore from
+// the newest backup.
+var _ = Describe("{IssueDeleteOfIncrementalBackupsAndRestore}", func() {
+	var (
+		credName                 string
+		clusterUid               string
+		cloudCredUID             string
+		fullBackupName           string
+		restoreName              string
+		backupLocationUID        string
+		customBackupLocationName string
+		incrementalBackupName    string
+		restoreNames             []string
+		cloudCredUidList         []string
+		namespaceMapping         map[string]string
+		appContexts              []*scheduler.Context
+		contexts                 []*scheduler.Context
+		clusterStatus            api.ClusterInfo_StatusInfo_Status
+	)
+	labelSelectors := make(map[string]string)
+	backupNames := make([]string, 0)
+	incrementalBackupNames := make([]string, 0)
+	incrementalBackupNames2 := make([]string, 0)
+	var bkpNamespaces = make([]string, 0)
+	backupLocationMap := make(map[string]string)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("IssueDeleteOfIncrementalBackupsAndRestore",
+			"Issue delete of incremental backups and try to restore the newest backup", nil, 58056)
+		log.InfoD("Deploy applications")
+
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+
+	It("Issue delete of incremental backups and try to restore the newest backup", func() {
+		providers := getProviders()
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+		})
+
+		Step("Adding Credentials and Registering Backup Location", func() {
+			log.InfoD("Creating cloud credentials and backup location")
+			for _, provider := range providers {
+				cloudCredUID = uuid.New()
+				cloudCredUidList = append(cloudCredUidList, cloudCredUID)
+				backupLocationUID = uuid.New()
+				credName := fmt.Sprintf("autogenerated-cred-%v", time.Now().Unix())
+				CreateCloudCredential(provider, credName, cloudCredUID, orgID)
+				log.InfoD("Created Cloud Credentials with name - %s", credName)
+				customBackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
+				backupLocationMap[backupLocationUID] = customBackupLocationName
+				err := CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
```

Comment:
How about updating backupLocationMap after CreateBackupLocation? If CreateBackupLocation location fails, CleanupCloudSettingsAndClusters in JustAfterEach might throw an error saying backupLocation does not exist.